tests/network/hotplug.go

```diff
@@ -0,0 +1,311 @@
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	expect "github.com/google/goexpect"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/framework/checks"
+	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/libwait"
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+var _ = SIGDescribe("nic-hotplug", func() {
+	const (
+		bridgeName  = "supadupabr"
+		ifaceName   = "iface1"
+		networkName = "skynet"
+		vmIfaceName = "eth1"
+	)
+
+	var virtClient kubecli.KubevirtClient
+
+	BeforeEach(func() {
+		var err error
+
+		virtClient, err = kubecli.GetKubevirtClient()
+		util.PanicOnError(err)
+	})
+
+	Context("a running VMI with the default number of PCI slots", func() {
+		var vmi *v1.VirtualMachineInstance
+
+		BeforeEach(func() {
+			vmi = setupVMI(virtClient, libvmi.NewAlpineWithTestTooling(libvmi.WithMasqueradeNetworking()...))
+			Expect(
+				createBridgeNetworkAttachmentDefinition(virtClient, util.NamespaceTestDefault, networkName, bridgeName),
+			).To(Succeed())
+			Expect(assertHotpluggedIfaceDoesNotExist(vmi, vmIfaceName)).To(Succeed())
+
+			Expect(
+				virtClient.VirtualMachineInstance(vmi.GetNamespace()).AddInterface(
+					vmi.GetName(),
+					addIfaceOptions(networkName, ifaceName),
+				),
+			).To(Succeed())
+
+			Eventually(func() []v1.VirtualMachineInstanceNetworkInterface {
+				return vmiCurrentInterfaces(virtClient, vmi.GetNamespace(), vmi.GetName())
+			}, 30*time.Second).Should(
+				WithTransform(
+					CleanMACAddressesFromStatus,
+					ConsistOf(interfaceStatusFromInterfaceNames(ifaceName))))
+		})
+
+		It("can be hotplugged a network interface", func() {
+			Expect(assertHotpluggedIfaceExists(vmi, vmIfaceName)).To(Succeed())
+		})
+
+		It("cannot hotplug multiple network interfaces for a q35 machine type by default", func() {
+			By("hotplugging the second interface")
+			const secondHotpluggedIfaceName = "iface2"
+			Expect(
+				virtClient.VirtualMachineInstance(vmi.GetNamespace()).AddInterface(
+					vmi.GetName(),
+					addIfaceOptions(networkName, secondHotpluggedIfaceName),
+				),
+			).To(Succeed())
+			Eventually(func() []corev1.Event {
+				events, err := virtClient.CoreV1().Events(vmi.GetNamespace()).List(context.TODO(), metav1.ListOptions{})
+				ExpectWithOffset(1, err).NotTo(HaveOccurred())
+				return events.Items
+			}, 30*time.Second).Should(
+				WithTransform(
+					filterVMISyncErrorEvents,
+					ContainElement(noPCISlotsAvailableError())))
+		})
+
+		It("can migrate a VMI with hotplugged interfaces", func() {
+			checks.SkipIfMigrationIsNotPossible()
+
+			migration := tests.NewRandomMigration(vmi.Name, vmi.Namespace)
+			migrationUID := tests.RunMigrationAndExpectCompletion(virtClient, migration, tests.MigrationWaitTime)
+			tests.ConfirmVMIPostMigration(virtClient, vmi, migrationUID)
```

Comment:
Can we go further here? It would be nice to test if the network is still attached and usable after the migration.